File: app/src/main/java/com/goranch/publicapis/ui/food/details/DetailRecipeView.java
package com.goranch.publicapis.ui.food.details;

import com.goranch.publicapis.api.model.food.nutrition.Food;

public interface DetailRecipeView {

    void showProgress();

    void hideProgress();

    void openWebView(String url);

    void getSingleRecipe(String recipeId);

    void onItemClicked(Food item);

    String getNutritionText(Food food);

    void showHTTPError();
}

File: app/src/main/java/com/goranch/publicapis/ui/food/details/DetailRecipeView.kt
package com.goranch.publicapis.ui.food.details

import com.goranch.publicapis.api.model.food.nutrition.Food

interface DetailRecipeView {

    fun showProgress()

    fun hideProgress()

    fun openWebView(url: String)

    fun getSingleRecipe(recipeId: String)

    fun onItemClicked(item: Food)

    fun getNutritionText(food: Food): String

    fun showHTTPError()
}
